src/amchart/CommonSerial.js

```diff
@@ -228,30 +247,40 @@
         this._chart.categoryAxis.color = this.xAxisFontColor();
         this._chart.categoryAxis.titleColor = this.xAxisTitleFontColor();
         this._chart.categoryAxis.titleFontSize = this.xAxisTitleFontSize();
-        //this._chart.titles = [];
 
-        this._chart.categoryAxis.labelFunction = function(d) {
-            return d;
-        };
+        this._chart.categoryAxis.showFirstLabel = this.showFirstLabel();
+        this._chart.categoryAxis.showLastLabel = this.showLastLabel();
+
+        this._chart.categoryAxis.equalSpacing = this.equalSpacing();
+
+        //this._chart.titles = [];
 
         switch(this.xAxisType()) {
             case "time":
-                this._chart.categoryAxis.type = "date";
                 this._chart.categoryAxis.parseDates = true;
-                //this._chart.valueAxes[0].minPeriod = "hh";
+                this._chart.categoryAxis.minPeriod = this.axisMinPeriod() ? this.axisMinPeriod() : undefined;
                 this._chart.categoryAxis.logarithmic = false;
+
+                if (this.xAxisTickFormat()) {
+                    this.dataFormatter = d3.time.format(this.xAxisTickFormat());
+                } else if (this.xAxisTypeTimePattern()) {
+                    this.dataFormatter = d3.time.format(this.xAxisTypeTimePattern());
+                } else {
+                    this.dataFormatter =  function(v) { return v; };
+                }
```

Comment:
I wouldn't mix my "data pattern parser" (string->date) and my display formatter (date->string) up like this.